lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -1077,6 +1085,25 @@ def build_accessible_object_scope(associations, options = {})
       end
     end
 
+    def use_accessible_objects_function?(options)
+      PolicyMachineStorageAdapter.postgres? &&
+        options[:direct_only] &&
+        options[:ignore_prohibitions] &&
+        options[:fields]&.one?
+    end
+
+    # Performance optimized function for PostgreSQL
+    def accessible_objects_function(user_id, operation, options)
+      accessible_objects_for_operations_function(user_id, Array.wrap(operation), options).values.flatten
+    end
+
+    # Performance optimized function for PostgreSQL
+    def accessible_objects_for_operations_function(user_id, operations, options)
+      operation_names = operations.map { |o| operation_identifier(o) }
+      accessible_map = operation_names.index_with([])
```

Comment:
this will have all the operation names point to the same array (i just did the same thing in another PR). block implementation will be invoked once for each operation name, so will be a different array for each,.
```suggestion
      accessible_map = operation_names.index_with { [] }
```

Reply:
Good catch!